apis/v1alpha1/gateway_types.go

```diff
@@ -249,6 +249,39 @@ type Listener struct {
 	// is required if the Protocol field is "HTTPS" or "TLS" and
 	// ignored otherwise.
 	//
+	// The association of SNIs to Certificate defined in TLSConfig is
+	// defined based on the value of HostnameMatchType for this listener:
+	// - "Domain": Certificate should be used used for the domain and its
+	//   first-level subdomains.
+	// - "Exact": Certificate should be used for the domain only.
+	// - "Any": Certificate in TLSConfig is the default certificate to use.
+	//
+	// The GatewayClass must use the most specific available certificate.
```

Comment:
Maybe GatewayClass be Gateway or implementation here?

Reply:
We follow GatewayClass in a lot of places and that's what I borrowed from here. Should I change it in this case or keep as is?

Reply:
This specific use feels more tied to implementation than class, but this does not need to block the PR.